### PR TITLE
Refactor messaging system for XRPL protocol endpoints

### PR DESCRIPTION
**Pull request by ThibautBremand**

### CODE REVIEW

File: packages/extension/src/components/pages/SignTransaction/SignTransaction.tsx [hunk 1]
```diff
@@ -109,24 +111,37 @@ export const SignTransaction: FC = () => {
...
+      }
+
+      setParams({
+        id,
+        txParam: parsedTransaction
+      });
+    };
+
+    fetchData();
   }, [wallet?.publicAddress]);
```
Comment (ThibautBremand) [inline packages/extension/src/components/pages/SignTransaction/SignTransaction.tsx:145]: obsolete

File: packages/api/src/helpers/extensionMessaging.ts [hunk 2]
```diff
@@ -112,10 +112,6 @@ const serializeMessage = (msg: APIMessages): any => {
     modifiedMsg.payload.flags = JSON.stringify(modifiedMsg.payload.flags);
   }
 
-  if (typeof modifiedMsg.payload?.transaction === 'object') {
```
Comment (FlorianBouron) [inline packages/api/src/helpers/extensionMessaging.ts:115]: Why did you remove this?

File: packages/extension/cypress/e2e/sign_transaction.cy.ts [hunk 3]
```diff
@@ -1,9 +1,14 @@
 import { Network, NETWORK } from '@gemwallet/constants';
 
-describe('Sign Transaction', () => {
-  // deepcode ignore NoHardcodedPasswords: password used for testing purposes
-  const PASSWORD = 'SECRET_PASSWORD';
+import { navigate } from '../utils/navigation';
+
+const URL =
+  'http://localhost:3000?sign-transaction&storageKey=1693425372955.3833&id=210401828&requestMessage=undefined&sign=transaction';
```
Comment (FlorianBouron) [inline packages/extension/cypress/e2e/sign_transaction.cy.ts:6]:
```suggestion
  'http://localhost:3000/sign-transaction&storageKey=1693425372955.3833&id=210401828&requestMessage=undefined&sign=transaction';
```

File: packages/extension/src/chromeServices/background/index.ts [hunk 4]
```diff
@@ -426,27 +426,25 @@ chrome.runtime.onMessage.addListener(
...
-        parameter: PARAMETER_SUBMIT_TRANSACTION,
-        receivingMessage: 'RECEIVE_SUBMIT_TRANSACTION/V3',
-        errorPayload: {
-          type: ResponseType.Reject,
-          result: undefined
-        }
-      });
+      const { payload } = message;
+      try {
+        sendInMemoryMessage({
```
Comment (FlorianBouron) [inline packages/extension/src/chromeServices/background/index.ts:431]: I think that this function is not properly named.
```suggestion
        sendMessageInMemory({
```

File: packages/extension/src/chromeServices/background/index.ts [hunk 5]
```diff
@@ -426,27 +426,25 @@ chrome.runtime.onMessage.addListener(
...
+          payload,
+          parameter: PARAMETER_SUBMIT_TRANSACTION,
+          receivingMessage: 'RECEIVE_SUBMIT_TRANSACTION/V3',
+          sender
+        });
+      } catch (e) {}
```
Comment (FlorianBouron) [inline packages/extension/src/chromeServices/background/index.ts:437]: Tbh if it fails should we try again?
Btw what would be the reasons for this promise to fail?

File: packages/extension/src/components/pages/SubmitTransaction/SubmitTransaction.tsx [hunk 6]
```diff
@@ -45,6 +47,14 @@ export const SubmitTransaction: FC = () => {
     params.txParam?.Fee ?? null
   );
 
+  const queryString = window.location.search;
+  const urlParams = new URLSearchParams(queryString);
```
Comment (FlorianBouron) [inline packages/extension/src/components/pages/SubmitTransaction/SubmitTransaction.tsx:51]:
```suggestion
  const urlParams = new URLSearchParams(window.location.search);
```

File: packages/extension/src/components/pages/SignTransaction/SignTransaction.tsx [hunk 7]
```diff
@@ -50,8 +51,14 @@ export const SignTransaction: FC = () => {
     },
     params.txParam?.Fee ?? null
   );
-  const { getCurrentWallet } = useWallet();
-  const wallet = getCurrentWallet();
+
+  const queryString = window.location.search;
+  const urlParams = new URLSearchParams(queryString);
```
Comment (FlorianBouron) [inline packages/extension/src/components/pages/SignTransaction/SignTransaction.tsx:56]: Same comment than below about the queryString

File: packages/extension/src/components/pages/SignTransaction/SignTransaction.tsx [hunk 8]
```diff
@@ -109,25 +116,23 @@ export const SignTransaction: FC = () => {
     const queryString = window.location.search;
     const urlParams = new URLSearchParams(queryString);
```
Comment (FlorianBouron) [inline packages/extension/src/components/pages/SignTransaction/SignTransaction.tsx:117]: Same comment than below about the queryString

File: packages/extension/src/components/pages/SubmitBulkTransactions/SubmitBulkTransactions.tsx [hunk 9]
```diff
@@ -154,52 +156,30 @@ export const SubmitBulkTransactions: FC = () => {
     const queryString = window.location.search;
     const urlParams = new URLSearchParams(queryString);
```
Comment (FlorianBouron) [inline packages/extension/src/components/pages/SubmitBulkTransactions/SubmitBulkTransactions.tsx:157]: Same comment than below about the queryString

File: packages/extension/src/components/pages/SubmitTransaction/SubmitTransaction.tsx [hunk 10]
```diff
@@ -101,17 +111,23 @@ export const SubmitTransaction: FC = () => {
     const queryString = window.location.search;
     const urlParams = new URLSearchParams(queryString);
```
Comment (FlorianBouron) [inline packages/extension/src/components/pages/SubmitTransaction/SubmitTransaction.tsx:112]: Same comment than above about the queryString

File: packages/extension/src/components/pages/SubmitBulkTransactions/SubmitBulkTransactions.tsx [hunk 11]
```diff
@@ -67,6 +61,14 @@ export const SubmitBulkTransactions: FC = () => {
     null
   );
 
+  const queryString = window.location.search;
+  const urlParams = new URLSearchParams(queryString);
```
Comment (FlorianBouron) [inline packages/extension/src/components/pages/SubmitBulkTransactions/SubmitBulkTransactions.tsx:65]: Same comment than below with the: queryString

File: packages/extension/src/chromeServices/background/index.ts (hunk 5 above)
Comment (ThibautBremand) [reply packages/extension/src/chromeServices/background/index.ts:437]: Error on the user's browser side that we do not know about (permissions, etc). idk what we can do in this case
